src/utransport.rs

```diff
@@ -16,9 +16,23 @@ use std::hash::{Hash, Hasher};
 use std::ops::Deref;
 use std::sync::Arc;
 
-use crate::{UMessage, UStatus, UUri};
+use crate::{UCode, UMessage, UStatus, UUri};
 
-/// `UListener` is the uP-L1 interface that provides a means to create listeners which are registered to [`UTransport`]
+/// A factory for URIs representing this uEntity's resources.
+///
+/// Implementations may use arbitrary mechanisms to determine the information that
+/// is necessary for creating URIs, e.g. environment variables, configuration files etc.
+pub trait LocalUriProvider: Send + Sync {
```

Comment:
Is there a specific usage scenario that we're aware of, where this trait needs to be Send+Sync?

Reply:
I assume that an implementation of this trait will be passed into an implementation of e.g. RpcClient. If we want RpcClient to be used from different threads concurrently, then LocalUriProvider being Send + Sync will allow the RpcClient to simply invoke its functions without bothering about synchronization.

Reply:
Ok thanks. I will implement this asap, because I've got equivalent code in my usub project already and want to see what this looks like from that perspective.